src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -94,21 +94,16 @@ public sealed class ResolvePackageAssets : TaskBase
         public bool EnsureRuntimePackageDependencies { get; set; }
 
         /// <summary>
-        /// Specifies whether to validate that the version of the implicit platform package in the assets
-        /// file matches the version specified by <see cref="ExpectedPlatformPackageVersion"/>
+        /// Specifies whether to validate that the version of the implicit platform packages in the assets
+        /// file matches the version specified by <see cref="ImplicitPlatformPackages"/>
         /// </summary>
         public bool VerifyMatchingImplicitPackageVersion { get; set; }
 
         /// <summary>
-        /// Identifier for implicitly referenced platform package.  If set, then an error will be generated if the
-        /// version of that package from the assets file does not match the <see cref="ExpectedPlatformPackageVersion"/>
+        /// Implicitly referenced platform packages.  If set, then an error will be generated if the
+        /// version of the specified packages from the assets file does not match the expected versions.
         /// </summary>
-        public string ImplicitPlatformPackageIdentifier { get; set; }
-
-        /// <summary>
-        /// Expected version of the <see cref="ImplicitPlatformPackageIdentifier"/>
-        /// </summary>
-        public string ExpectedPlatformPackageVersion { get; set; }
+        public ITaskItem[] ImplicitPlatformPackages { get; set; }
```

Comment:
I would include "Expected" in the name of this property.  Perhaps `ExpectedPackages` or `ExpectedPlatformPackages`.